deploy/scripts/src/authentication/test.ts

```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
+            params: { tags: { name: 'B02_SignIn_06_SMSMFA_EnterPassword' } }
           }), { isStatusCode200, ...pageContentCheck('Check your phone') }))
 
       sleep(1)
 
-      res = group('B01_SignIn_08_SMSMFA_EnterOTP POST', () =>
-        timeRequest(() => {
-          const response = res.submitForm({
-            fields: { code: credentials.phoneOTP },
-            params: { tags: { name: 'B01_SignIn_08_SMSMFA_EnterOTP' } }
-          })
-          acceptNewTerms = (response.body as string).includes('terms of use update')
-          return response
-        }, { isStatusCode200, 'verify page content': r => acceptNewTerms || (r.body as string).includes('User information') }))
+      group('B02_SignIn_07_SMSMFA_EnterOTP POST', () => {
+        res = timeRequest(() => res.submitForm({
+          fields: { code: credentials.phoneOTP },
+          params: {
+            redirects: 1,
+            tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_01_AuthCall' }
+          }
+        }),
+        { isStatusCode302 })
+        res = timeRequest(() => http.get(res.headers.Location, {
+          redirects: 0,
+          tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_02_OIDCCall' }
+        }),
+        { isStatusCode302 })
+
+        rpStubCheck = (res.headers.Location).includes('auth-stub')
+        acceptNewTerms = (res.headers.Location).includes('signin')
+
+        if (rpStubCheck) {
+          res = timeRequest(() => http.get(res.headers.Location, {
+            tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_03_RPStub' }
+          }),
+          { isStatusCode200, ...pageContentCheck('User information') })
+        } else if (acceptNewTerms) {
+          res = timeRequest(() => http.get(res.headers.Location, {
+            tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_03_AuthAcceptTerms' } // pragma: allowlist secret
+          }),
+          { isStatusCode200, ...pageContentCheck('terms of use update') })
+        }
+      })
       break
     }
   }
 
   if (acceptNewTerms) {
-    res = group('B01_SignIn_09_AcceptTermsConditions POST', () =>
+    res = group('B02_SignIn_08_AcceptTermsConditions POST', () =>
       timeRequest(() =>
         res.submitForm({
           fields: { termsAndConditionsResult: 'accept' },
-          params: { tags: { name: 'B01_SignIn_09_AcceptTermsConditions' } }
+          params: { tags: { name: 'B02_SignIn_08_AcceptTermsConditions' } }
         }), { isStatusCode200, ...pageContentCheck('User information') }))
   }
 
+  iterationsCompleted.add(1)
+
   // 25% of users logout
   if (Math.random() <= 0.25) {
     sleep(1)
 
-    res = group('B01_SignIn_10_Logout POST', () =>
+    res = group('B02_SignIn_09_Logout POST', () =>
       timeRequest(() => res.submitForm({
-        params: { tags: { name: 'B01_SignIn_10_Logout' } }
+        params: { tags: { name: 'B02_SignIn_09_Logout' } }
       }), { isStatusCode200, ...pageContentCheck('Successfully signed out') }))
   }
-  iterationsCompleted.add(1)
```

Comment:
Keep this at the end of the function

Reply:
Same as previous comment - The logout step is purely a stub call so this metric has been moved before that step to provide an accurate view of the number of sign-ups and sign-ins.

Reply:
The logout step has a redirect call to Orchestrator. I have updated this to split the stub and OIDC calls in the logout step.
Also, the iterationCompleted metric has been moved at the end of the function.